theano/gof/vm.py

```diff
@@ -406,6 +407,8 @@ def __call__(self, output_subset=None):
 
         for k in self.storage_map:
             compute_map[k][0] = (k.owner is None)
+            if self.callback_input and compute_map[k][0]:
+                self.callback_input(k, self.storage_map[k][0])
```

Comment:
This behave as the old code. I just taught that this also check constant. I would be fine to only test the inputs of the fgraph and not constant. Otherwise, update the docstring around line 688 to tell this.